### PR TITLE
Libcap bump 2.56

### PR DESCRIPTION
**Pull request by SoapGentoo**

### CODE REVIEW

File: sys-libs/libcap/libcap-2.56.ebuild [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+DESCRIPTION="POSIX 1003.1e capabilities"
+HOMEPAGE="https://sites.google.com/site/fullycapable/"
+SRC_URI="https://www.kernel.org/pub/linux/libs/security/linux-privs/libcap2/${P}.tar.xz"
+
+# it's available under either of the licenses
+LICENSE="|| ( GPL-2 BSD )"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux"
+IUSE="pam static-libs"
```
Comment (thesamesam) [inline sys-libs/libcap/libcap-2.56.ebuild:16]: captree!

**Review by thesamesam — CHANGES_REQUESTED** [review]

**Review by mgorny — APPROVED** [review]
LGTM. For a special value of 'good'.

**Comment by SoapGentoo** [comment]
```
--- libcap-2.55-r1.ebuild
+++ libcap-2.56.ebuild
@@ -13,24 +13,24 @@
 LICENSE="|| ( GPL-2 BSD )"
 SLOT="0"
 KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux"
-IUSE="pam static-libs"
+IUSE="pam static-libs tools"
 
 # While the build system optionally uses gperf, we don't DEPEND on it because
 # the build automatically falls back when it's unavailable.  #604802
 PDEPEND="pam? ( sys-libs/pam[${MULTILIB_USEDEP}] )"
 DEPEND="${PDEPEND}
 	sys-kernel/linux-headers"
-
-# Requires test suite being run as root (via sudo)
-RESTRICT="test"
+BDEPEND="tools? ( dev-lang/go )"
 
 PATCHES=(
-	"${FILESDIR}"/${PN}-2.55-build-system-fixes.patch
+	"${FILESDIR}"/${PN}-2.56-build-system-fixes.patch
 	"${FILESDIR}"/${PN}-2.38-no_perl.patch
 	"${FILESDIR}"/${PN}-2.25-ignore-RAISE_SETFCAP-install-failures.patch
 	"${FILESDIR}"/${PN}-2.21-include.patch
 )
 
+QA_FLAGS_IGNORED="sbin/captree" # go binaries don't use LDFLAGS
+
 src_prepare() {
 	default
 	multilib_copy_sources
@@ -38,19 +38,22 @@
 
 run_emake() {
 	local args=(
+		AR="$(tc-getAR)"
+		CC="$(tc-getCC)"
+		OBJCOPY="$(tc-getOBJCOPY)"
+		RANLIB="$(tc-getRANLIB)"
 		exec_prefix="${EPREFIX}"
 		lib_prefix="${EPREFIX}/usr"
 		lib="$(get_libdir)"
 		prefix="${EPREFIX}/usr"
 		PAM_CAP="$(usex pam yes no)"
 		DYNAMIC=yes
-		GOLANG=no
+		GOLANG="$(multilib_native_usex tools yes no)"
 	)
 	emake "${args[@]}" "$@"
 }
 
 src_configure() {
-	tc-export AR CC OBJCOPY RANLIB
 	tc-export_build_env BUILD_CC
 	multilib-minimal_src_configure
 }
@@ -59,6 +62,10 @@
 	run_emake
 }
 
+multilib_src_test() {
+	run_emake test
+}
+
 multilib_src_install() {
 	# no configure, needs explicit install line #444724#c3
 	run_emake DESTDIR="${D}" install
@@ -69,9 +76,8 @@
 		rm "${ED}"/usr/$(get_libdir)/lib{cap,psx}.a || die
 	fi
 
-	if [[ -d "${ED}"/usr/$(get_libdir)/security ]] ; then
-		rm -r "${ED}"/usr/$(get_libdir)/security || die
-	fi
+	# install pam plugins ourselves
+	rm -rf "${ED}"/usr/$(get_libdir)/security || die
 
 	if use pam ; then
 		dopammod pam_cap/pam_cap.so
```

File: sys-libs/libcap/metadata.xml [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
-</maintainer>
-<upstream>
-	<remote-id type="cpe">cpe:/a:libcap:libcap</remote-id>
-</upstream>
+	<maintainer type="project">
+		<email>base-system@gentoo.org</email>
+		<name>Gentoo Base System</name>
+	</maintainer>
+	<use>
+		<flag name="tools">Install captree, which requires dev-lang/go.</flag>
```
Comment (ionenwks) [inline sys-libs/libcap/metadata.xml:9]: Don't we *usually* avoid `.` in these? Also missing `<pkg></pkg>`

That aside, looks go to me.

**Review by thesamesam — APPROVED** [review]

**Review by ionenwks — APPROVED** [review]

**Pull request closed without merging**